### PR TITLE
feat(api): get payment order by id

### LINKED ISSUE #4: ASAUIW to obtain the payment order detail
## US
ASAUIW to get payment order detail 

## Description
We want to allow the developer to obtain the payment order detail 

## Additional context
Scenario 1:

```
Given... a SDK user 
When... 
Then...
And...
```

## Resources & Design

```typescript
public async getById(id: string): Promise<PaymentOrderResponse>
```

[SCHEMA](https://developers.mymoid.com/api-reference#/schemas/PaymentOrder)
[API Endpoint](https://developers.mymoid.com/api-reference#/operations/GetPaymentOrderByPaymentOrderId)
 

## Acceptance criteria

- [ ] ...

## Technical tasks

- [ ] Add `getById(id)` public async method with mapped respond into PaymentOrdersEndpoints class.
- [ ] Handle response errors.
- [ ] Create Types.
- [ ] Add examples 

## Definition of Ready:
- [ ] Acceptance criteria have been defined for the story.
- [ ] Acceptance criteria are clear and understood by the whole team.
- [ ] Story has been estimated and is within a certain size limit.
- [ ] Story is clear and understood by the entire team.
- [ ] UI designer has created mockups or fully designed screens for the story.
- [ ] All dependencies, internal or external, have been resolved.

## Definition of Done:
- [ ] All technical tasks are done.
- [ ] All acceptance criteria are met.
- [ ] Changes have been reviewed.
- [ ] CI tests have run successfully.
- [ ] Documentation has been updated.
- [ ] Release Notes have been updated.

### PR DESCRIPTION
**Pull request by juanchojbarroso-mymoid**
Closes #4

Changes proposed in this pull request:

- Get payment order by id
- Add an example

@juanchojbarroso-mymoid

### CODE REVIEW

File: examples/nextjs/app/ui/payment-orders/table.tsx [hunk 1]
```diff
@@ -28,7 +29,7 @@ export default function PaymentOrdersTable({
         {paymentOrders?.map((paymentOrder) => (
           <tr
             key={paymentOrder.paymentOrderId}
-            className="border-b border-neutral-600"
+            className="group/item border-b border-neutral-600 hover:bg-gray-800"
```
Comment (mmarine-mymoid) [inline examples/nextjs/app/ui/payment-orders/table.tsx:32]:
```suggestion
            className="group/item border-b border-neutral-600 hover:bg-neutral-900 transition-colors"
```

File: examples/nextjs/app/ui/payment-orders/table.tsx [hunk 2]
```diff
@@ -56,6 +57,24 @@ export default function PaymentOrdersTable({
                 minute: 'numeric'
               })}
             </td>
+            <td className="group/edit invisible whitespace-nowrap px-4 py-3 text-sm group-hover/item:visible">
```
Comment (mmarine-mymoid) [inline examples/nextjs/app/ui/payment-orders/table.tsx:60]:
```suggestion
            <td className="group/edit opacity-0 whitespace-nowrap px-4 py-3 text-sm group-hover/item:opacity-100 transition-opacity">
```

In my opinion, these suggestions improve the user experience.

File: .changeset/rude-panthers-tease.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+"nextjs": minor
+"@mymoid/api": minor
+---
+
+Get payment order by id
```
Comment (mmarine-mymoid) [inline .changeset/rude-panthers-tease.md:6]:
```suggestion
We're thrilled to introduce a powerful enhancement to streamline your experience: the ability to effortlessly retrieve payment orders by their unique identifiers! Say goodbye to unnecessary complexity and hello to efficiency with our new "Get payment order by id" feature.
```
Is crazy, just to keep it in mind. 😎

File: examples/nextjs/app/[slug]/page.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+import { Button } from '../ui/button'
+import MainHeader from '../ui/main-header'
+import PaymentOrderDetail from '../ui/payment-orders/detail'
+
+export default async function Page({ params }: { params: { slug: string } }) {
+  const paymentOrderId = params.slug
+  return (
+    <>
+      <MainHeader title={`Payment order`} subtitle={`ID: ${paymentOrderId}`}>
+        <form
```
Comment (mmarine-mymoid) [inline examples/nextjs/app/[slug]/page.tsx:11]: This button is not strictly necessary. 😎

Comment (mmarine-mymoid) [reply examples/nextjs/app/[slug]/page.tsx:11]: In order to keep the code readable.

File: examples/nextjs/app/[slug]/page.tsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
+import { revalidatePath } from 'next/cache'
+import { Button } from '../ui/button'
+import MainHeader from '../ui/main-header'
+import PaymentOrderDetail from '../ui/payment-orders/detail'
+
+export default async function Page({ params }: { params: { slug: string } }) {
+  const paymentOrderId = params.slug
+  return (
+    <>
+      <MainHeader title={`Payment order`} subtitle={`ID: ${paymentOrderId}`}>
```
Comment (mmarine-mymoid) [inline examples/nextjs/app/[slug]/page.tsx:10]:
```suggestion
      <MainHeader title={`Payment Order`} subtitle={`ID: ${paymentOrderId}`}>
```

File: examples/nextjs/app/ui/payment-orders/detail.tsx [hunk 6]
```diff
@@ -0,0 +1,29 @@
+import { mymoid } from '@/app/lib/mymoid-api'
+import { formatCurrency } from '@/app/lib/utils'
+import { Field } from '../Field'
+
+export default async function PaymentOrderDetail({ id }: { id: string }) {
+  const paymentOrder = await mymoid.paymentOrders.getById(id)
```
Comment (mmarine-mymoid) [inline examples/nextjs/app/ui/payment-orders/detail.tsx:6]: 🚀

File: examples/nextjs/app/ui/payment-orders/table.tsx [hunk 7]
```diff
@@ -56,6 +57,24 @@ export default function PaymentOrdersTable({
...
+                <svg
+                  xmlns="http://www.w3.org/2000/svg"
+                  fill="none"
+                  viewBox="0 0 24 24"
+                  stroke-width="1.5"
+                  stroke="currentColor"
+                  className="w-4 h-4"
+                >
+                  <path
+                    stroke-linecap="round"
```
[the remark this answers is not in the document]
Comment (juanchojbarroso) [reply examples/nextjs/app/ui/payment-orders/table.tsx:71]: Done!

File: examples/nextjs/app/ui/payment-orders/table.tsx [hunk 8]
```diff
@@ -56,6 +57,24 @@ export default function PaymentOrdersTable({
...
+                    stroke-linejoin="round"
```
[the remark this answers is not in the document]
Comment (juanchojbarroso) [reply examples/nextjs/app/ui/payment-orders/table.tsx:72]: Done!

File: examples/nextjs/app/ui/payment-orders/table.tsx (hunk 1 above)
Comment (juanchojbarroso) [reply examples/nextjs/app/ui/payment-orders/table.tsx:32]: Much better!

**Comment by juanchojbarroso-mymoid** [comment]
@mmarine-mymoid I have already made all the necessary changes. Could you check this out?

**Review by mmarine-mymoid — APPROVED** [review]

**Pull request merged**